base/expr.jl

```diff
@@ -109,13 +109,17 @@ evaluates expressions in that module.
 Core.eval
 
 """
-    @eval
+    @eval [m,] ex
 
-Evaluate an expression and return the value.
+Evaluate an expression with values interpolated into it using `eval`.
+If two arguments are provided, the first is the module to evaluate in.
 """
 macro eval(x)
     :($(esc(:eval))($(Expr(:quote,x))))
 end
+macro eval(m, x)
+    :($(esc(:eval))($(esc(m)),$(Expr(:quote,x))))
```

Comment:
That `:eval` probably shouldn't be in `esc`. Also true of the other macro (it should probably instead splice in the value of `current_module()` as the first argument to `eval`).